Scheduler/REST/CSharpClient/Program.cs

```diff
@@ -1,906 +1,729 @@
-
-        public static ICredentials Credentials
-        {
-            get { return _credentials; }
-            set { _credentials = value; }
-        }
-
-        /// <summary>
-        /// this callback disables enforcement of x509 certificat trust chains.
-        /// it enables the use of self-signed certs.
-        /// 
-        /// if you desire to enforce trust chains you must not register this callback.
-        /// </summary>
-        /// <param name="obj"></param>
-        /// <param name="cert"></param>
-        /// <param name="chain"></param>
-        /// <param name="err"></param>
-        /// <returns></returns>
-        public static bool DisableTrustChainEnforecment(object obj, X509Certificate cert, X509Chain chain, SslPolicyErrors err)
-        {
-            return true;
-        }
-
-        public static HttpWebRequest CreateRequest(string uri, string method)
-        {
-            Console.WriteLine($"{method} \"{uri}\"");
-
-            HttpWebRequest req = HttpWebRequest.Create(uri) as HttpWebRequest;
-
-            req.Credentials = RequestFactory.Credentials;
-
-            // here we add the required version header
-            req.Headers["api-version"] = "2012-11-01.4.0";
-
-            req.Method = method;
-
-            return req;
-        }
-    }
-}
+// Copyright © Microsoft Corporation.  All Rights Reserved.
+// This code released under the terms of the
+// MICROSOFT LIMITED PUBLIC LICENSE version 1.1 (MS-LPL, http://go.microsoft.com/?linkid=9791213.)
+//
+//Copyright (C) Microsoft Corporation.  All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+
+/*
+ * The following sample sources no longer require the REST Starter Kit.
+ *
+ * In addition, the variable "credentialsAreCachedOnHN" can be set:
+ *      true: if the credentials (ntlm or basic) are cached on the head node
+ *      false: otherwise.
+*/
+
+namespace RestClient
+{
+    public class Program
+    {
+        const string ContinuationHeader = "x-ms-continuation-";
+        const string QueryIdQueryParam = "QueryId";
+        const string QueryIdHeader = ContinuationHeader + QueryIdQueryParam;
+
+        static void Main(string[] args)
+        {
+            MainAsync(args).GetAwaiter().GetResult();
+        }
+
+        static void ShowHelp()
+        {
+            string help = @"
+Usage:
+{0} -c <server name> -u <user name> -p <password> [-C]
+
+Options:
+-c HPC server name to connect to.
+-u Name of a HPC user on the server.
+-p Password of the user.
+-C The switch specifies that the user credentail is cached on the HPC server. It's not by default.
+";
+            Console.WriteLine(String.Format(help, System.Diagnostics.Process.GetCurrentProcess().ProcessName));
+        }
+
+        static async Task MainAsync(string[] args)

```

Comment:
As it's a `C# 7.1` feature which is not available until `VS 2017.3`, we should notice users via `readme` this requirement.

Reply:
I'm not using the new "async main". I'm just using a normal async method `MainAsync` and call it from the normal `Main`.